File: Mage/Map/Cache/GeoPackageTableCacheOverlay.m
//
//  GeoPackageTableCacheOverlay.m
//  MAGE
//
//  Created by Brian Osborn on 12/18/15.
//  Copyright © 2015 National Geospatial Intelligence Agency. All rights reserved.
//

#import "GeoPackageTableCacheOverlay.h"

@interface GeoPackageTableCacheOverlay ()

@property (strong, nonatomic) NSString * geoPackage;
@property (nonatomic) int count;
@property (nonatomic) int minZoom;
@property (nonatomic) int maxZoom;

@end

@implementation GeoPackageTableCacheOverlay

-(instancetype) initWithName: (NSString *) name andGeoPackage: (NSString *) geoPackage andCacheName: (NSString *) cacheName andType: (enum CacheOverlayType) type andCount: (int) count andMinZoom: (int) minZoom andMaxZoom: (int) maxZoom{
    self = [super initWithName:name andCacheName:cacheName andType:type andSupportsChildren:false];
    if(self){
        self.geoPackage = geoPackage;
        self.count = count;
        self.minZoom = minZoom;
        self.maxZoom = maxZoom;
    }
    return self;
}

-(void) removeFromMap: (MKMapView *) mapView{
    if(self.tileOverlay != nil){
        [mapView removeOverlay:self.tileOverlay];
        self.tileOverlay = nil;
    }
}

-(NSString *) getGeoPackage{
    return self.geoPackage;
}

-(int) getCount{
    return self.count;
}

-(int) getMinZoom{
    return self.minZoom;
}

-(int) getMaxZoom{
    return self.maxZoom;
}

@end

File: Mage/Map/Cache/GeoPackageTableCacheOverlay.swift
//
//  GeoPackageTableCacheOverlay.m
//  MAGE
//
//  Created by Brian Osborn on 12/18/15.
//  Copyright © 2015 National Geospatial Intelligence Agency. All rights reserved.
//

class GeoPackageTableCacheOverlay: CacheOverlay {
    var tileOverlay: MKTileOverlay?
    var parent: CacheOverlay?
    var geoPackage: String
    var count: Int
    var minZoom: Int
    var maxZoom: Int
    
    init(name: String, geoPackage: String, cacheName: String, type: CacheOverlayType, count: Int, minZoom: Int, maxZoom: Int) {
        self.geoPackage = geoPackage
        self.count = count
        self.minZoom = minZoom
        self.maxZoom = maxZoom
        super.init(name: name, cacheName: cacheName, type: type, supportsChildren: false)
        self.isChild = true
    }
    
    override func getParent() -> CacheOverlay? {
        return parent
    }
    
    override func removeFromMap(mapView: MKMapView) {
        if tileOverlay != nil {
            mapView.removeOverlay(tileOverlay!)
            tileOverlay = nil
        }
    }
}

